src/net.cpp

```diff
@@ -455,6 +455,8 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
         connect_to.push_back(addrConnect);
     }
 
```

Comment:
The function `AddI2PSessionsIfNeeded` is called without checking if `m_i2p_sam_session` is null, which can lead to unnecessary session creation and potential resource leaks.